drivers/ccs811/ccs811.c

```diff
@@ -0,0 +1,684 @@
+/*
+ * Copyright (C) 2018 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ccs811
+ * @brief       Device Driver for AMS CCS811 digital gas sensor
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @file
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "log.h"
+#include "xtimer.h"
+
+#include "ccs811_regs.h"
+#include "ccs811.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+/**
+ * Internal macro definitions
+ */
+
+#define ASSERT_PARAM(cond) \
+    if (!(cond)) { \
+        DEBUG("[ccs811] %s: %s\n", \
+              __func__, "parameter condition (" # cond ") not fulfilled"); \
+        assert(cond); \
+    }
+
+#define DEBUG_DEV(f, d, ...) \
+    DEBUG("[ccs811] %s dev=%d addr=%02x: " f "\n", \
+          __func__, d->params.i2c_dev, d->params.i2c_addr, ## __VA_ARGS__)
+
+#define ERROR_DEV(f, d, ...) \
+    LOG_ERROR("[ccs811] dev=%d addr=%x: " f "\n", \
+              d->params.i2c_dev, d->params.i2c_addr, ## __VA_ARGS__)
+
+/**
+ * Internal type declarations
+ */
+
+typedef struct {
+    uint8_t reserved_1 : 2;
+    uint8_t int_thresh : 1; /**< interrupt if new ALG_RESULT_DAT crosses on of the thresholds */
+    uint8_t int_datardy: 1; /**< interrupt if new sample is ready in ALG_RESULT_DAT  */
+    uint8_t drive_mode : 3; /**< mode number binary coded */
+} ccs811_meas_mode_reg_t;
+
+/**
+ * forward declaration of functions for internal use only
+ */
+static int _reg_read(const ccs811_t *dev, uint8_t reg, uint8_t *data, uint32_t len);
+static int _reg_write(const ccs811_t *dev, uint8_t reg, uint8_t *data, uint32_t len);
+static int _check_error_status(const ccs811_t *dev);
+static int _error_code(const ccs811_t *dev, uint8_t err_reg);
+static int _is_available(const ccs811_t *dev);
+
+int ccs811_init(ccs811_t *dev, const ccs811_params_t *params)
+{
+    ASSERT_PARAM(dev != NULL);
+    ASSERT_PARAM(params != NULL);
+
+    /* init sensor data structure */
+    dev->params = *params;
+
+    int res = CCS811_OK;
+
+#if MODULE_CCS811_FULL
+    if (dev->params.reset_pin != GPIO_UNDEF &&
```

Comment:
I'm also wondering if this block could also be part of the basic driver. Since the reset pin is not mandatory, no error should be raised it the reset pin is GPIO_UNDEF.

Reply:
The same question as above. Size of code and number of GPIOs required.

Reply:
merged